packages/grid/data-grid/src/tests/components.DataGrid.test.tsx

```diff
@@ -49,6 +50,70 @@ describe('<DataGrid /> - Components', () => {
     });
   });
 
+  describe('componentsProps', () => {
+    it('should pass the props from componentsProps.cell to the cell', () => {
+      render(
+        <div style={{ width: 300, height: 500 }}>
+          <DataGrid
+            {...baselineProps}
+            hideFooter
+            disableVirtualization
+            componentsProps={{ cell: { 'data-name': 'foobar' } }}
+          />
+        </div>,
+      );
+      expect(getCell(0, 0)).to.have.attr('data-name', 'foobar');
+    });
+
+    it('should pass the props from componentsProps.row to the row', () => {
+      render(
+        <div style={{ width: 300, height: 500 }}>
+          <DataGrid
+            {...baselineProps}
+            hideFooter
+            disableVirtualization
+            componentsProps={{ row: { 'data-name': 'foobar' } }}
+          />
+        </div>,
+      );
+      expect(getRow(0)).to.have.attr('data-name', 'foobar');
+    });
+  });
+
+  describe('components', () => {
+    it('should render the cell with the component given in components.Cell', () => {
+      render(
+        <div style={{ width: 300, height: 500 }}>
+          <DataGrid
+            {...baselineProps}
+            hideFooter
+            disableVirtualization
+            components={{
+              Cell: ({ rowIndex, colIndex }) => (
```

Comment:
Maybe a relevant question but what is the difference between providing a cell like this and doing it via the `renderCell` in the `colDef`? What if you do both?

Reply:
We can wait for the feedback from the community. In reality, I don't see an use case for having to override the cell or row (maybe to render `<tr>` and `<td>`), even on the demo I didn't have to do this. This PR is more for the `componentsProps` which will be used to re-add events I want to remove later.

Reply:
OK so for you the idea is to theoratically handle the `props.components.row` and `props.components.cell` but to talk little about it in the doc because it is really an edge case ?

Reply:
Yes, I added `components.Cell` for convention but users will rarely need to replace the cell.